### ISSUE TITLE
Dragging into inventory bypasses setCancelled

### ISSUE THREAD

**Issue by Meex-0** [stefvanschie/IF#86]
[https://youtu.be/Oq2GrtH9gK4](url)
This video shows that dragging items into an inventory where `gui.setOnTopClick(e -> e.setCancelled(true));` is used gets bypassed.

I suggest adding:
```java
gui.setOnTopDrag(event -> {});
gui.setOnBottomDrag(event -> {});
gui.setOnGlobalDrag(event -> {});
```
so you can cancel Inventory drags separately from Inventory clicks.

Usage example:
```java
gui.setOnGlobalDrag(event -> e.setCancelled(true));
```
cancels all inventory drags.

**Comment by stefvanschie** [comment]
How should be determined whether a drag is a top drag or bottom drag? Looking at the javadocs for `InventoryDragEvent`, there seems to be no way to check on which slot the drag starts. `InventoryDragEvent#getInventorySlots` is a `Set`, so there is no guarantee on order either.

**Comment by Meex-0** [comment]
Already answered you on discord but I will post an example of my answer on github as well if anyone else needs help.
Note: I am not experienced with java so this can contain inefficient code, etc.

```java
InventoryView view = e.getView();
Set<Integer> slotSet = e.getRawSlots();
ArrayList<Inventory> inventoryList = new ArrayList<>();

for (Integer slot : slotSet) {
	inventoryList.add(view.getInventory(slot));
}

Boolean containsTop = inventoryList.contains(view.getTopInventory());
Boolean containsBottom = inventoryList.contains(view.getBottomInventory());

if (containsBottom && containsTop) {
	System.out.println("Contains bottom and top inventory!");
}
else if (containsTop) {
	System.out.println("Contains top inventory!");
}
else if (containsBottom) {
	System.out.println("Contains bottom inventory!");
}
```

**Comment by stefvanschie** [comment]
Thank you, this has been added and will be available in 0.9.0.
